File: in_Java/0086 Partition List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode partition(ListNode head, int x) {
        ListNode a = new ListNode(0);
        ListNode b = new ListNode(0);
        ListNode h = head;
        ListNode newa = a;
        ListNode newb = b;
        ListNode next;
        while(h != null){
            next = h.next;
            if(h.val < x){
                a.next = h;
                a = a.next;
                a.next = null;
            }else{
                b.next = h;
                b = b.next;
                b.next = null;
            }
            h = next;
        }
        a.next = newb.next;
        return newa.next;
    }
}

File: in_Python/0086 Partition List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def partition(self, head, x):
        """
        :type head: ListNode
        :type x: int
        :rtype: ListNode
        """

        d1 = ListNode(0)
        d2 = ListNode(0)
        dd1 = d1
        dd2 = d2
        while head != None:
            if head.val < x:
                d1.next = ListNode(head.val)
                d1 = d1.next
            else:
                d2.next = ListNode(head.val)
                d2 = d2.next
            head = head.next
        d1.next = dd2.next
        return dd1.next